package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -0,0 +1,306 @@
+            :meth:`~MDAnalysis.core.AtomGroup.AtomGroup.select_atoms`
+            This selection of atoms is used to calculate the RMSD between
+            different frames. Water should be excluded.
+        metric : function, optional
+            Maps two numpy arrays to a float, is positive definite and
+            symmetric, Default: metric is set to rms.rmsd().
+        cutoff : float, optional
+            Specify a given cutoff for metric values to be considered equal,
+            Default: 1EO-5
+        weights : array, optional
+            Weights to be given to coordinates for metric calculation
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+        """
+
+        self._u = u
+        traj = self._u.trajectory
+        self.atoms = self._u.select_atoms(select)
+        self._metric = metric
+        self._cutoff = cutoff
+        self._weights = weights
+        # remember that this must be called before referencing self.nframes
+        self._setup_frames(traj, start, stop, step)
+
+    def _prepare(self):
+        logger.info('numframes {0}'.format(self.nframes))
+        self.d_matrix = np.zeros((self.nframes, self.nframes))
+
+    def _single_frame(self):
+        traj_index = self._ts.frame
+        i_ref = self.atoms.positions - self.atoms.center_of_mass()
+        # diagonal entries need not be calculated due to metric(x,x) == 0 in
+        # theory, _ts not updated properly. Possible savings by setting a
+        # cutoff for significant decimal places to sparsify matrix
+        for j in range(self.stop-1, self._ts.frame-1, -self.step):
+            self._ts = self._u.trajectory[j]
+            j_ref = self.atoms.positions-self.atoms.center_of_mass()
+            dist = self._metric(i_ref, j_ref, weights=self._weights)
+            self.d_matrix[traj_index, j] = dist if dist > self._cutoff else 0
+            self.d_matrix[j, traj_index] = self.d_matrix[traj_index, j]
+
+    def store(self, filename):
+        pass
+
+    def load(self, filename):
+        pass
+
+
+class DiffusionMap(DistMatrix):
+    """Non-linear dimension reduction method
+
+    Dimension reduction with diffusion map of the structures in the universe.
+
+    Attributes
+    ----------
+    atoms : AtomGroup
+        Selected atoms in trajectory subject to dimension reduction
+    d_matrix : array
+        Array of all possible ij metric distances between frames in trajectory
+    eigenvalues: array
+        Eigenvalues of the diffusion map
+    eigenvectors: array
+        Eigenvectors of the diffusion map
+        The second and higher eigenvectors ev[i+1,:] represent the i-th slowest
+        collective coordinates.
+    """
+
+    def __init__(self, DistMatrix, epsilon=epsilon_constant, weights=None,
```

Comment:
Will do. I spent the day today reading literature and writing a comprehensive blog post for diffusion maps. After some thinking and hard work I am ready to push out the blog post and a final product for diffusion maps tomorrow.